src/java/org/apache/cassandra/config/BitRate.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.config;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.google.common.primitives.Ints;
+
+/**
+ * Represents a bit rate type used for cassandra configuration. It supports the opportunity for the users to be able to
+ * add units to the confiuration parameter value. (CASSANDRA-15234)
+ */
+public final class BitRate
+{
+    /**
+     * The Regexp used to parse the rate provided as String in cassandra.yaml.
+     */
+    private static final Pattern BIT_RATE_UNITS_PATTERN = Pattern.compile("^(\\d+)(MiB/s|mib/s|MiB/S|KiB/s|kib/s|KiB/S|B/s|b/s|B/S)$");
+
+    private final long quantity;
+
+    private final BitRateUnit unit;
+
+    public BitRate(String value)
+    {
+        //parse the string field value
+        Matcher matcher = BIT_RATE_UNITS_PATTERN.matcher(value);
+
+        if (!matcher.find())
+        {
+            throw new IllegalArgumentException("Invalid bit rate: " + value + " Accepted units: MiB/s, KiB/s, B/s.");
+        }
+
+        quantity = Long.parseLong(matcher.group(1));
+        unit = BitRateUnit.fromSymbol(matcher.group(2));
+    }
+
+    private BitRate(long quantity, BitRateUnit unit)
+    {
+        this.quantity = quantity;
+        this.unit = unit;
+    }
+
+    /**
+     * Creates a {@code BitRate} of the specified amount of bits per second.
+     *
+     * @param bitsPerSeconds the amount of bits per second
+     * @return a {@code BitRate}
+     */
+    public static BitRate inBitsPerSeconds(long bitsPerSeconds)
```

Comment:
```suggestion
    public static BitRate inBitsPerSecond(long bitsPerSecond)
```

Reply:
Fixed this typo everywhere I saw it in the class, thank you (commit later)